USAGE.md

```diff
@@ -107,3 +112,18 @@ echo $response->statusCode();
 echo $response->body();
 echo $response->headers();
 ```
+<a name="exceptions"></a>
+## Exception Handling
+
+```
+try {
+    $response = $client->setThrowException(true)->api_keys()->_($WRONG_api_key)->patch($request_body);
+}
+catch(ClientException $e) {
+    var_dump(
+        $e->getErrors(),
+        $e->getHttpStatus(),
+        $e->getShortTrace());
+    exit;
```

Comment:
I'm not sure that the exit code is zero  when catching the `ClientException`.

The exception is triggered and the exit code is not zero.

Reply:
this is very sensitive situation. Let's imagine I have several servers, which are DEVELOPING, TESTING, STAGING, PRODUCTION and all of them have own error_reporting_level. 

And nobody knows which errors are set via error_reporting. To check it I read bitmask.

```php
	$errorReporting = convertBitMask(error_reporting());

	private function convertBitMask($decimal) {
		$scan = 1;
		$result = [];
		while($decimal >= $scan) {
			if($decimal & $scan)
				$result[] = $scan;
			$scan <<= 1;
		}

		return $result;
	}
```

In this situation we can't set any specific error code, even according to http://php.net/manual/en/errorfunc.constants.php, because different environments can have different error levels.

So if you somebody need special error code for sendgrid API, he can set it in this manner

```php
$client->setThrowException(true, E_USER_WARNING);
```

https://github.com/sendgrid/php-http-client/pull/131/files#diff-d25a5d86da2d487fe7787e90709a5ecbR350

Reply:
https://github.com/sendgrid/php-http-client/pull/131/commits/5f9df0e40abc0d8255262181137ab7151fee7011

some explanation